pkgs/applications/editors/vscode/vscode-oss/product.nix

```diff
@@ -0,0 +1,408 @@
+{ lib }:
+{
+  nameShort = "Code - OSS";
+  nameLong = "Code - OSS";
+  applicationName = "code-oss";
+  dataFolderName = ".vscode-oss";
+  win32MutexName = "vscodeoss";
+  licenseName = "MIT";
+  licenseUrl = "https://github.com/microsoft/vscode/blob/main/LICENSE.txt";
+  serverLicenseUrl = "https://github.com/microsoft/vscode/blob/main/LICENSE.txt";
+  serverGreeting = [ ];
+  serverLicense = [ ];
+  serverLicensePrompt = "";
+  serverApplicationName = "code-server-oss";
+  serverDataFolderName = ".vscode-server-oss";
+  tunnelApplicationName = "code-tunnel-oss";
+  win32DirName = "Microsoft Code OSS";
+  win32NameVersion = "Microsoft Code OSS";
+  win32RegValueName = "CodeOSS";
+  win32x64AppId = "{{D77B7E06-80BA-4137-BCF4-654B95CCEBC5}";
+  win32arm64AppId = "{{D1ACE434-89C5-48D1-88D3-E2991DF85475}";
+  win32x64UserAppId = "{{CC6B787D-37A0-49E8-AE24-8559A032BE0C}";
+  win32arm64UserAppId = "{{3AEBF0C8-F733-4AD4-BADE-FDB816D53D7B}";
+  win32AppUserModelId = "Microsoft.CodeOSS";
+  win32ShellNameShort = "C&ode - OSS";
+  win32TunnelServiceMutex = "vscodeoss-tunnelservice";
+  win32TunnelMutex = "vscodeoss-tunnel";
+  darwinBundleIdentifier = "com.visualstudio.code.oss";
+  linuxIconName = "code-oss";
+  licenseFileName = "LICENSE.txt";
+  reportIssueUrl = "https://github.com/microsoft/vscode/issues/new";
+  nodejsRepository = "https://nodejs.org";
+  urlProtocol = "code-oss";
+  webviewContentExternalBaseUrlTemplate = "https://{{uuid}}.vscode-cdn.net/insider/ef65ac1ba57f57f2a3961bfe94aa20481caca4c6/out/vs/workbench/contrib/webview/browser/pre/";
+
+  extensionAllowedBadgeProviders = [
+    "api.bintray.com"
+    "api.travis-ci.com"
+    "api.travis-ci.org"
+    "app.fossa.io"
+    "badge.buildkite.com"
+    "badge.fury.io"
+    "badge.waffle.io"
+    "badgen.net"
+    "badges.frapsoft.com"
+    "badges.gitter.im"
+    "badges.greenkeeper.io"
+    "cdn.travis-ci.com"
+    "cdn.travis-ci.org"
+    "ci.appveyor.com"
+    "circleci.com"
+    "cla.opensource.microsoft.com"
+    "codacy.com"
+    "codeclimate.com"
+    "codecov.io"
+    "coveralls.io"
+    "david-dm.org"
+    "deepscan.io"
+    "dev.azure.com"
+    "docs.rs"
+    "flat.badgen.net"
+    "gemnasium.com"
+    "githost.io"
+    "gitlab.com"
+    "godoc.org"
+    "goreportcard.com"
+    "img.shields.io"
+    "isitmaintained.com"
+    "marketplace.visualstudio.com"
+    "nodesecurity.io"
+    "opencollective.com"
+    "snyk.io"
+    "travis-ci.com"
+    "travis-ci.org"
+    "visualstudio.com"
+    "vsmarketplacebadge.apphb.com"
+    "www.bithound.io"
+    "www.versioneye.com"
+  ];
+  extensionAllowedBadgeProvidersRegex = [
+    "^https:\\/\\/github\\.com\\/[^/]+\\/[^/]+\\/(actions\\/)?workflows\\/.*badge\\.svg"
+  ];
+  extensionEnabledApiProposals = {
```

Comment:
This seems rather aggressive to put this into nix code. Can't we read that from the downloaded source file?

Reply:
> Can't we read that from the downloaded source file?

The source file is rather empty, and these items listed mainly comes from VSCodium community.

https://github.com/microsoft/vscode/blob/main/product.json